util-core/src/main/scala/com/twitter/concurrent/AsyncStream.scala

```diff
@@ -465,7 +469,11 @@ final class AsyncStream[+A] private (private val underlying: Future[LazySeq[A]])
 }
 
 object AsyncStream {
-  import LazySeq.{Cons, Empty, One}
+  private case object Empty extends AsyncStream[Nothing]
+  private case class Embed[A](fas: Future[AsyncStream[A]]) extends AsyncStream[A]
+  private case class FromFuture[A](fa: Future[A]) extends AsyncStream[A]
+  private case class Cons[A](fa: Future[A], more: () => AsyncStream[A])
```

Comment:
I overrode it initially because the tail argument was call by name, which toString would have forced, evaluating it prematurely. Now that it's a function, toString won't force it. We could change it back to preserve the '?' but it might be confusing given that argument is now a function.